apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx

```diff
@@ -959,6 +991,289 @@ export const renderConnectYourDevice = ({
+  isSwap = false,
+}: {
+  onContinue: () => void;
+  deviceId: DeviceModelId;
+  isSwap?: boolean;
+}) => {
+  return (
+    <Box
+      flow={6}
+      mx={20}
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        justifyContent: "center",
+        ...(isSwap ? { height: "100%" } : {}),
+      }}
+    >
+      <Box
+        flow={3}
+        style={{
+          width: "100%",
+        }}
+      >
+        <CircleIcon>
+          <Icons.WarningFill size="L" color="palette.warning.c70" />
+        </CircleIcon>
+        <Text fontWeight="600" textAlign="center" fontSize={24}>
+          <Trans
+            i18nKey={`DeviceDeprecation.warning.title`}
+            values={{
+              device: getDeviceModel(deviceId).productName,
+            }}
+          />
+        </Text>
+        <Text color="palette.text.shade60" textAlign="center" fontWeight="500" fontSize={14}>
+          <Trans i18nKey={`DeviceDeprecation.warning.subtitle`} />
+        </Text>
+      </Box>
+      <Box
+        flow={4}
+        style={{
+          width: "100%",
+        }}
+      >
+        <ButtonV3
+          variant="main"
+          size="large"
+          onClick={(e: { preventDefault: () => void }) => {
+            e.preventDefault();
+            openURL("https://shop.ledger.com/pages/ledger-nano-s-upgrade-program");
+          }}
+        >
+          <Trans i18nKey={`DeviceDeprecation.update`} />
+        </ButtonV3>
+        <ButtonV3
+          size="large"
+          variant="shade"
+          onClick={() => {
+            onContinue();
+          }}
+        >
+          <Trans i18nKey={`DeviceDeprecation.warning.continue`} />
+        </ButtonV3>
+        <Link
+          href="https://support.ledger.com/article/Ledger-Nano-S-Limitations"
+          color="palette.text.shade60"
+          onClick={e => {
+            e.preventDefault();
+            openURL("https://support.ledger.com/article/Ledger-Nano-S-Limitations");
+          }}
+        >
+          <Trans i18nKey={`DeviceDeprecation.learnMore`} />
+        </Link>
+      </Box>
+    </Box>
+  );
+};
+
+export const DeprecationWarning = ({
+  coinName,
+  date,
+  onContinue,
+  deviceId,
+  isSwap = false,
+}: {
+  coinName: string;
+  date: string;
+  onContinue: () => void;
+  deviceId: DeviceModelId;
+  isSwap?: boolean;
+}) => {
+  const dispatch = useDispatch();
+  const [isChecked, setIsChecked] = useState(false);
+  const dateFormatted = useCalendarFormatted(new Date(date));
+  const toggleCheck = useCallback(() => {
+    setIsChecked(prev => !prev);
+  }, []);
+
+  const handleContinue = () => {
+    if (isChecked) {
+      dispatch(deprecateWarningReminder(coinName));
+    }
+    onContinue();
+  };
+
+  return (
+    <Box
+      flow={6}
+      mx={20}
+      alignSelf="center"
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        justifyContent: "center",
```

Comment:
There's a component `Flex` from `"@ledgerhq/react-ui"` that does what you need. And 99% of the time you don't need to use inline styles in LL.
Don't hesitate to check the rest of the codebase to see how components from this UI library are supposed to be used (thousands of examples).

And when in doubt, you can ask the live-hub team. It's usually much better if you ask other teams for help during your implementation process rather than by asking for a PR review after 1 month of work. It will be much smoother and save a lot of time for everyone involved.